### PR TITLE
feat/workflow

### PR DESCRIPTION
**Pull request by martin-olivier**
## Description

This pull request provides the following changes on the workflow:

- Build and tests now runs on linux, macOS and windows
- CI now checks the conventional commits rule
- CI now runs `clippy-rs` on source code

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Tests
- [ ] Documentation
- [X] Other (for changes that might not fit in any category)

## Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [X] I have tested this code
- [X] I have added sufficient documentation in the code
- [X] I have added the right flag to this PR

### CODE REVIEW

**Review by 0xpanoramix — APPROVED** [review]

File: .github/workflows/ConventionalCommits.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: CI
```
Comment (Breigner01) [inline .github/workflows/ConventionalCommits.yml:1]: Maybe a name other than CI would be nice to quickly distinguish in the actions tab which one is which

File: .github/workflows/Linter.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: CI
```
Comment (Breigner01) [inline .github/workflows/Linter.yml:1]: Maybe a name other than CI would be nice to quickly distinguish in the actions tab which one is which

**Review by Breigner01 — COMMENTED** [review]
Minor details otherwise that's good

**Review by TomChv — COMMENTED** [review]
That's a nice improvement, some comments:

1 - `CI` -> `Continuous Integration`
2 - Merge yamls in a single one since they are all related
3 - Do not trigger CI is the code or dependencies hasn't change
4 - Add dependencies on job, IMO you should not build the app if pass do not test and you should not test the app if the lint failed. That would do `lint -> test -> build`. WDYT?

File: .github/workflows/Linter.yml (hunk 2 above)
Comment (martin-olivier) [reply .github/workflows/Linter.yml]: Done

**Comment by martin-olivier** [comment]
> CI -> Continuous Integration
Merge yamls in a single one since they are all related
Do not trigger CI is the code or dependencies hasn't change

Done in [fd74766](https://github.com/Megalotron/Sniffer/pull/1/commits/fd747662bd1c0cc4b6e91dfd69f459acefb96f5f)

> Add dependencies on job, IMO you should not build the app if pass do not test and you should not test the app if the lint failed. That would do lint -> test -> build. WDYT?

How can i make that ?

**Comment by TomChv** [comment]
> How can i make that ?

Use `needs` keywords, let keep it for another PR

**Review by TomChv — DISMISSED** [review]

**Review by 0xpanoramix — APPROVED** [review]

**Comment by TomChv** [comment]
Sign your commit and we merge it

**Pull request closed without merging**